unitTests/ui/SimpleFilterTest.ts

```diff
@@ -1,227 +1,327 @@
 import * as Mock from '../MockEnvironment';
 import { ISimpleFilterOptions, SimpleFilter } from '../../src/ui/SimpleFilter/SimpleFilter';
 import { Simulate } from '../Simulate';
-import { $$ } from '../../src/utils/Dom';
+import { $$, Dom } from '../../src/utils/Dom';
 import { BreadcrumbEvents, IPopulateBreadcrumbEventArgs } from '../../src/events/BreadcrumbEvents';
 import { FakeResults } from '../Fake';
 import { analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
 
 export function SimpleFilterTest() {
+  const enableClearButtonOptions = [false, true, null];
+  const simpleFilterOptions = {
+    field: '@field',
+    maximumNumberOfValues: 5,
+    title: 'FooTitleBar',
+    values: ['foo', 'bar'],
+    availableSorts: ['score'],
+    sortCriteria: 'score',
+    valueCaption: {
+      bar: 'baz',
+      gmailmessage: 'Gmail Message',
+      lithiummessage: 'Lithium Message',
+      youtubevideo: 'Youtube Video',
+      message: 'Message'
+    }
+  };
+
   describe('SimpleFilter', () => {
     let aSimpleFilter: Mock.IBasicComponentSetup<SimpleFilter>;
 
-    beforeEach(() => {
-      aSimpleFilter = Mock.optionsComponentSetup<SimpleFilter, ISimpleFilterOptions>(SimpleFilter, <ISimpleFilterOptions>{
-        field: '@field',
-        maximumNumberOfValues: 5,
-        title: 'FooTitleBar',
-        values: ['foo', 'bar'],
-        availableSorts: ['score'],
-        sortCriteria: 'score',
-        valueCaption: {
-          bar: 'baz',
-          gmailmessage: 'Gmail Message',
-          lithiummessage: 'Lithium Message',
-          youtubevideo: 'Youtube Video',
-          message: 'Message'
-        }
+    enableClearButtonOptions.forEach(enableClearButton => {
```

Comment:
It's kind of weird that we are testing all component `options` inside that loop, but that loop is only meant to tests variation on `enableClearButton`.

What if we want to add a new test for different variation on another option ? We'd have to do some weird changes.

I'd change it so that we have a `describe` block that checks for all options individually, and then inside that block, add another `describe` block for the variations on `enableClearButtonOptions `